include/deal.II/base/index_set.h

```diff
@@ -1218,6 +1232,44 @@ IndexSet::IndexSet (const size_type size)
 
 
 
+#ifdef DEAL_II_WITH_CXX11
+
+inline
+IndexSet::IndexSet (IndexSet &&is)
+  :
+  ranges (std::move(is.ranges)),
+  is_compressed (is.is_compressed),
+  index_space_size (is.index_space_size),
+  largest_range (is.largest_range)
+{
+  is.clear ();
+  is.index_space_size = 0;
+  is.largest_range = numbers::invalid_unsigned_int;
+
+  compress ();
```

Comment:
I'm glad you agree, I wasn't entirely sure about this myself. Besides the reason you give, it also seemed consistent with other member functions of `IndexSet`, which often compress the inputs and outputs.